internal/db/db_blueprints.go

```diff
@@ -27,6 +51,44 @@ const (
 	sqlDeleteBlueprint = `DELETE FROM blueprints WHERE id = $1 AND org_id = $2 AND account_number = $3`
 )
 
+func (db *dB) GetBlueprintComposes(orgId string, blueprintId uuid.UUID, since time.Duration, limit, offset int, ignoreImageTypes []string) ([]ComposeEntryWithVersion, int, error) {
+	ctx := context.Background()
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Release()
+	result, err := conn.Query(ctx, sqlGetBlueprintComposes, orgId, blueprintId, since, ignoreImageTypes, limit, offset)
+	defer result.Close()
+
+	if err != nil {
+		return nil, 0, err
+	}
```

Comment:
You have to defer close call after `nil` check, because in case of an error `result` will actually be `nil` = panic.